core/src/main/scala/org/apache/spark/sql/SnappyStrategies.scala

```diff
@@ -81,13 +81,20 @@ private[sql] trait SnappyStrategies {
       case _ => Nil
     }
 
+    def isLocalJoin(plan: LogicalPlan): Boolean = plan match {
+      case ExtractEquiJoinKeys(joinType, _, _, _, left, right) =>
+        (canBuildRight(joinType) && canLocalJoin(right)) ||
+            (canBuildLeft(joinType) && canLocalJoin(left))
```

Comment:
Yes that was the idea. The LocalHashMap can be built for exchange results too where we want to skip de-linking (or rather enforce linking by LinkPartitionToBuckets). We could add more precise checks later which check for collocated join cases too, but right now played it safe so for collocated joins the partitions used will be same as buckets.